### PR TITLE
anonymous function callback inside an object parameter

### PR DESCRIPTION
**Pull request by tuxan**
I added support to use anonymous function as callback inside an object.
So it is possible to do next:

```javascript
bound.testCallbackFromObject({
  foo: "bar",
  callback: function(text) {
    console.log(text);
  }
});
```

It is possible to merge this feature to the project?

This is my first pull request :)

### CODE REVIEW

File: CefSharp.BrowserSubprocess.Core/TypeUtils.h [hunk 1]
```diff
@@ -28,7 +29,18 @@ namespace CefSharp
         /// </summary>
         /// <param name="obj">The V8 value that should be converted.</param>
         /// <returns>A corresponding .NET object.</returns>
-        static Object^ ConvertFromCef(CefRefPtr<CefV8Value> obj);
+		static Object^ ConvertFromCef(CefRefPtr<CefV8Value> obj);
```
Comment (amaitland) [inline CefSharp.BrowserSubprocess.Core/TypeUtils.h]: Remove this overload, just pass in `nullptr`. Formatting is out, spaces not tabs.

File: CefSharp.BrowserSubprocess.Core/TypeUtils.cpp [hunk 2]
```diff
@@ -148,7 +148,11 @@ namespace CefSharp
         throw gcnew Exception(String::Format("Cannot convert '{0}' object from CLR to CEF.", type->FullName));
     }
 
-    Object^ TypeUtils::ConvertFromCef(CefRefPtr<CefV8Value> obj)
+	Object^ TypeUtils::ConvertFromCef(CefRefPtr<CefV8Value> obj) {
+		return TypeUtils::ConvertFromCef(obj, nullptr);
+	}
+
+	Object^ TypeUtils::ConvertFromCef(CefRefPtr<CefV8Value> obj, JavascriptCallbackRegistry^ callbackRegistry)
```
Comment (amaitland) [inline CefSharp.BrowserSubprocess.Core/TypeUtils.cpp]: Formattting.

File: CefSharp.BrowserSubprocess.Core/TypeUtils.cpp [hunk 3]
```diff
@@ -148,7 +148,11 @@ namespace CefSharp
         throw gcnew Exception(String::Format("Cannot convert '{0}' object from CLR to CEF.", type->FullName));
     }
 
-    Object^ TypeUtils::ConvertFromCef(CefRefPtr<CefV8Value> obj)
+	Object^ TypeUtils::ConvertFromCef(CefRefPtr<CefV8Value> obj) {
```
Comment (amaitland) [inline CefSharp.BrowserSubprocess.Core/TypeUtils.cpp]: Remove overload.

File: CefSharp.BrowserSubprocess.Core/TypeUtils.cpp [hunk 4]
```diff
@@ -195,6 +199,16 @@ namespace CefSharp
             return nullptr;
         }
 
+		if (obj->IsFunction())
```
Comment (amaitland) [inline CefSharp.BrowserSubprocess.Core/TypeUtils.cpp]: Formatting.

File: CefSharp.BrowserSubprocess.Core/TypeUtils.cpp [hunk 5]
```diff
@@ -195,6 +199,16 @@ namespace CefSharp
...
+		{
+			if (callbackRegistry != nullptr)
```
Comment (amaitland) [inline CefSharp.BrowserSubprocess.Core/TypeUtils.cpp]: Prefer `==` check over `!=` check.

```
if(callbackRegistry == nullptr)
{
  return nullptr;
}
```

File: CefSharp.Example/BoundObject.cs [hunk 6]
```diff
@@ -3,7 +3,10 @@
 // Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
 
 using System;
+using System.Collections;
+using System.Collections.Generic;
 using System.Linq;
+using System.Reflection;
```
Comment (amaitland) [inline CefSharp.Example/BoundObject.cs]: Is this using required?

**Comment by amaitland** [comment]
@tuxan Thanks for the contribution :+1: Please review your code changes using the `Files Changed` tab above. Your formatting is out, Spaces not tabs.

It's too late in the release cycle for this to be included in `51.0.0`. Can be included in the next major release.

**Comment by tuxan** [comment]
@amaitland I done the changes you reviewed.

**Comment by amaitland** [comment]
@tuxan Thanks :+1: Will try to take it for a spin sometime this coming week.

**Comment by amaitland** [comment]
@tuxan Merged with a few changes, nothing major :+1: 

https://github.com/cefsharp/CefSharp/commit/18793c68e9533565f77d56d2270e476359273c4d
https://github.com/cefsharp/CefSharp/commit/c8f6017fa576f61cdfc55240dd368059088733f3

**Comment by tuxan** [comment]
@amaitland Thanks :+1:

**Pull request merged**